server/child-process-runner/run-in-child-process.js

```diff
@@ -151,18 +184,12 @@ async function runInChildProcess(modulePath, runArguments, { timeout }) {
 
       // When a problem occurs when spawning the process or gets aborted
       child.on('error', (err) => {
+        // We should be able to just `reject(err)` without any special-case handling
+        // here since ideally, we expect the error to be whatever `signal.reason` we
+        // aborted with but `child_process.fork(...)` doesn't seem play nicely, see
+        // https://github.com/nodejs/node/issues/47814
         if (err.name === 'AbortError') {
-          const childErrorSummary = assembleErrorAfterChildExitsWithErrors(
-            childExitCode,
-            childErrors,
-            childStdErr
-          );
-          reject(
-            new RethrownError(
-              `Timed out while running ${modulePath} so we aborted the child process after ${timeout}ms. Any child errors? (${childErrors.length})`,
-              childErrorSummary
-            )
-          );
+          reject(abortController.signal.reason || err);
```

Comment:
Created https://github.com/nodejs/node/issues/47814 to track when we can stop doing this workaround